nsot/models.py

```diff
@@ -1718,6 +1719,99 @@ def to_dict(self):
         }
 
 
+class Protocol(Resource):
+    """
+    Represetation of a routing protocol running over a circuit
+    """
+
+    device = models.ForeignKey(
+        Device, db_index=True, null=False, related_name='protocols',
+        verbose_name='Device that this protocol is running on'
+    )
+    circuit = models.ForeignKey(
```

Comment:
If the above suggestion is implemented, this would need to be modified accordingly as well.